etc/scripts/release.hs

```diff
@@ -89,17 +92,32 @@ main =
                 gProjectRoot = "" -- Set to real value velow.
                 gBuildArgs = []
                 gCertificateName = Nothing
-                global0 = foldl (flip id) Global{..} flags
+                globalDefault = Global{..}
+
+            -- modify our default value of type Global with env vars and command line args
+            envOptions <- envOptionsIO
+            let globalWithEnv = envOptions globalDefault
+                globalWithArgsEnv = foldl (flip id) globalWithEnv flags
+
             -- Need to get paths after options since the '--arch' argument can effect them.
-            projectRoot' <- getStackPath global0 "project-root"
-            let global = global0
+            projectRoot' <- getStackPath globalWithArgsEnv "project-root"
+            let global = globalWithArgsEnv
                     { gProjectRoot = projectRoot' }
             return $ Just $ rules global args
   where
     getStackPath global path = do
       out <- readProcess stackProgName (stackArgs global ++ ["path", "--" ++ path]) ""
       return $ trim $ fromMaybe out $ stripPrefix (path ++ ":") out
 
+-- | Env Vars to set some of the command-line options found below.
+envOptionsIO :: IO (Global -> Global)
+envOptionsIO = do
+    envVars <- getEnvironment
+    let modifyByEnv global (k, v) = case k of
+          "STACK_RELEASE_GPG_KEY" -> global{gGpgKey = v}
```

Comment:
The name of the environment variable should be a constant (e.g. named `gpgKeyEnvVar`), similar to how `githubAuthTokenEnvVar` is a constant.